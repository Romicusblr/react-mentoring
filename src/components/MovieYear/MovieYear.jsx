import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './MovieYear.module.css';

function MovieYear({ year, className, ...other }) {
  return (
    <span
      className={classNames(style.movieYear, className)}
      {...other}
    >
      {year}
    </span>
  );
}

MovieYear.propTypes = {
  year: PropTypes.number.isRequired,
  className: PropTypes.string,
};

MovieYear.defaultProps = {
  className: '',
};

export default MovieYear;
