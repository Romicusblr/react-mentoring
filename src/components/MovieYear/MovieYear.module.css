.movieYear {
    border: 1px solid var(--color-text-other);
    padding: .25rem .5rem;
    border-radius: .25rem;
    color: var(--color-text-secondary);
}
