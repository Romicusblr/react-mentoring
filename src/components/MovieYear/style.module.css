.movieYear {
    border: 1px solid var(--color-gray);
    padding: .25rem .5rem;
    border-radius: .25rem;
    color: var(--color-lighter-gray);
}
