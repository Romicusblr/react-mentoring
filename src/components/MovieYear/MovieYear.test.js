import MovieYear from '.';

describe('MovieYear', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<MovieYear year={2000} />);
    expect(snap).toMatchSnapshot();
  });
});
