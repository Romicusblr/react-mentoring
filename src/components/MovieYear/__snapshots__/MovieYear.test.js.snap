// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieYear should match to snapshot 1`] = `
<span
  className="movieYear"
>
  2000
</span>
`;
